### PR TITLE
Feature/real time analyses

### PR DESCRIPTION
**Pull request by PerryShafran-NOAA**
## Pull Request Testing ##

- [X ] Describe testing already performed for this Pull Request:</br>

Daily runs of all stats file to generate daily stats
Rigorous testing of plots scripts

- [ ] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

1) Clone the fork to /lfs/h2/emc/vpppg/save/$USER and checkout the feature branch feature/real_time_analyses
2) For stats:  Go to the EVS/ecf/realtime_analyses/stats directory.  Execute the run files run_rtma_stats.sh, run_urma_stats.sh, and run_rtma_ru_stats.sh.  Each of these files runs 24 hours of analyses for their respective analysis.
3. Small stats files can be found in /lfs/h2/emc/vpppg/noscrub/$USER/evs/v1.0/stats/realtime_analyses/atmos.20221129/$anl/grid2obs.
4. Final stat files can be found in /lfs/h2/emc/vpppg/noscrub/$USER/evs/v1.0/stats/realtime_analyses/$anl.20221129
5. For plots:  Go to the EVS/ecf/realtime_analyses/plots directory.  Execute the file jevs_realtime_analyses_plots.ecf.
6. Individual plots can be found here:  /lfs/h2/emc/ptmp/$USER/EVS/00_rtma.
7. Final plot tar file can be found here:  /lfs/h2/emc/vpppg/noscrub/$USER/evs/v1.0/plots/atmos.20221129

- [X ] Has the code been checked to ensure that no errors occur during the execution? **[Yes or No]**

- [X ] Do these updates/additions include sufficient testing updates? **[Yes or No]**

- [ ] Please complete this pull request review by **[Fill in date]**.</br>

## Pull Request Checklist ##

- [X ] Review the source issue metadata (required labels, projects, and milestone).
- [X ] Complete the PR description above.
- [X ] Ensure the PR title matches the feature branch name.
- [ ] Check the following:
- [x]  Instructions provided on how to run
- [X ]  Developer's name is replaced by ${user} where necessary throughout the code
- [X ]  Check that the ecf file has all the proper definitions of variables
- [X ]  Check that the jobs file has all the proper settings of COMIN and COMOUT and other input variables
- [X]  Check to see that the output directory structure is followed
- [X ]  Be sure that you are not using MET utilities outside the METplus wrapper structure

- [ ] After submitting the PR, select **Development** issue with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue.

### CODE REVIEW

**Comment by ShelleyMelchior-NOAA** [comment]
Getting started on this today. I have checked out the branch feature/real_time_analyses and will begin the steps to run.

**Comment by ShelleyMelchior-NOAA** [comment]
The stats for rtma were successfully produced and output in COMOUTsmall and COMOUTfinal matches developer results.

Next up, urma and rtma_ru.

**Comment by ShelleyMelchior-NOAA** [comment]
The stats for rtma, rtma_ru, and urma were successfully produced. @PerryShafran-NOAA , please review the output in the respective COMOUT directories on cactus:
`/lfs/h2/emc/obsproc/noscrub/shelley.melchior/evs/v1.0/stats/realtime_analyses`

**Comment by PerryShafran-NOAA** [comment]
The stat files are now identical between your runs and my runs.

Perry

File: ecf/realtime_analyses/plots/jevs_realtime_analyses_plots.ecf [hunk 1]
```diff
@@ -1 +1,83 @@
-sample
+#PBS -N jevs_rtma_plots_00
```
Comment (ShelleyMelchior-NOAA) [inline ecf/realtime_analyses/plots/jevs_realtime_analyses_plots.ecf:1]: Curious about this line ... you have 3 different stats ecf scripts for each of the three MODELNAMEs. And those 3 ecf scripts use the MODELNAME in their job name. But for plots there is only one ecf script. Should the job name not be MODELNAME specific? It makes it seem like there are no plots for rtma_ru or urma. Maybe that is the case?

Comment (PerryShafran-NOAA) [reply ecf/realtime_analyses/plots/jevs_realtime_analyses_plots.ecf:1]: I could rename this to jevs_realtime_analyes_plots, like the name of the ecf file itself.  I don't think I even need the cycle time.  Plots for all three analyses are made, each one being analysis vs guess.  

Perry

**Comment by ShelleyMelchior-NOAA** [comment]
The `ush/real_time_analyses/plot_util.py` script had to be modified at line 1627.

`with open('/u/perry.shafran/test.pkl','wb') as f:`
replaced by
`with open('/u/shelley.melchior/test.pkl','wb') as f:`

This was a local change required to get through the testing of this PR. There will need to be an **_official resolution_** for this line in all `plot_util.py` scripts! Can this file reside in a non-user specific location, perhaps ush/ itself?

**Comment by ShelleyMelchior-NOAA** [comment]
I was able to run the plots processing and generate all of the png graphics as well as the final tar file on cactus.
`/lfs/h2/emc/ptmp/shelley.melchior/EVS/00_rtma/plots`
and
`/lfs/h2/emc/obsproc/noscrub/shelley.melchior/evs/v1.0/plots/realtime_analyses/atmos.20221207`

**Comment by ShelleyMelchior-NOAA** [comment]
I re-ran all of the testing for this PR on dogwood today (12/9/2022). There were no problems. Please review the output in the following locations:
`/lfs/h2/emc/obsproc/noscrub/shelley.melchior/evs/v1.0/stats/realtime_analyses`
`/lfs/h2/emc/obsproc/noscrub/shelley.melchior/evs/v1.0/plots/realtime_analyses`

**Comment by PerryShafran-NOAA** [comment]
All stat files are identical.

**Comment by ShelleyMelchior-NOAA** [comment]
Handled the conflict in `versions/run.ver`.

**Review by ShelleyMelchior-NOAA — APPROVED** [review]
This PR is approved for merge but one thing needs to be noted to be fixed before release to NCO.

The `ush/real_time_analyses/plot_util.py` script had to be modified at line 1627.

`with open('/u/perry.shafran/test.pkl','wb') as f:`
replaced by
`with open('/u/shelley.melchior/test.pkl','wb') as f:`

This was a local change required to get through the testing of this PR. There will need to be an **_official resolution_** for this line in all `plot_util.py` scripts! Can this file reside in a non-user specific location, perhaps ush/ itself?

**Pull request merged**